--- loky/backend/stdlib_py314_resource_tracker.py.ref	2026-08-20 12:38:59.671027411 +0200
+++ loky/backend/stdlib_py314_resource_tracker.py	2026-08-20 12:38:47.394820500 +0200
@@ -26,8 +26,9 @@
 
 import json
 
-from . import spawn
-from . import util
+# loky: fix imports
+from multiprocessing import spawn
+from multiprocessing import util
 
 __all__ = ['ensure_running', 'register', 'unregister']
 
@@ -383,15 +384,21 @@
 # popen_fork.Popen._launch() at once from clobbering eachothers intent.
 _fork_intent = threading.local()
 
+# loky: begin commented out section
+# Remove module-level globals and logic which
+# are unused and can cause confusing errors at interpreter exit because of
+# discrepancies in the code from vendored Python version and used Python
+# version
-_resource_tracker = ResourceTracker()
-ensure_running = _resource_tracker.ensure_running
-register = _resource_tracker.register
-unregister = _resource_tracker.unregister
-getfd = _resource_tracker.getfd
-
-# gh-146313: See _after_fork_in_child docstring.
-if hasattr(os, 'register_at_fork'):
-    os.register_at_fork(after_in_child=_resource_tracker._after_fork_in_child)
+# _resource_tracker = ResourceTracker()
+# ensure_running = _resource_tracker.ensure_running
+# register = _resource_tracker.register
+# unregister = _resource_tracker.unregister
+# getfd = _resource_tracker.getfd
+
+# # gh-146313: See _after_fork_in_child docstring.
+# if hasattr(os, 'register_at_fork'):
+#     os.register_at_fork(after_in_child=_resource_tracker._after_fork_in_child)
+# loky: end commented out section
 
 
 def _decode_message(line):
